### PR TITLE
playwright: Update to latest versions and add new page objects

### PR DESCRIPTION
**Pull request by ndoschek**
#### What it does

- Update `playwright` to the latest version
  - Update test configuration to adapt to latest version
  - Remove obsolete multiple config files and scripts
- Add page objects for OutputView, OutputChannel, Toolbar, MonacoEditor
- Extend tests by testing the new page objects
- Extend ExplorerView tests with compact folders
- Extend QuickCommand tests
- Update GETTING_STARTED.md and DEVELOPING.md
- Update build script to use ensure clean and lint are called
  - Ensure all necessary playwright dependencies are installed on initial build

Signed-off-by: Nina Doschek <ndoschek@eclipsesource.com>

#### How to test

- Run playwright tests via `yarn test:playwright` from root directory

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by ndoschek** [comment]
Hi @planger, as you already had an initial look on my changes, could you perhaps suggest additional reviewers for this PR? Thank you!

**Comment by planger** [comment]
@vince-fugnitto Would you maybe have time to look into this change? I've already reviewed it internally.
Marc has done some work on Playwright recently as well, but I've already asked him to review https://github.com/eclipse-theia/theia/pull/12381, so I don't want to overload him :-)

File: examples/playwright/src/theia-toolbar.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+import { ElementHandle } from '@playwright/test';
```
Comment (vince-fugnitto) [inline examples/playwright/src/theia-toolbar.ts:16]: minor: newline missing.

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me, I confirmed they pass locally and during CI :+1: 
I'm not sure if @marcdumais-work would like to take a look as well but given it works it is likely not necessary. We can keep an eye out if things get flaky.

**Comment by vince-fugnitto** [comment]
I was hoping to get https://github.com/eclipse-theia/theia/pull/12320 in first due to issues with the "out-of-date" CI check but it's fine.

**Comment by planger** [comment]
Oh I see... sorry about that! I saw that Nina rebased and all checks passed. So I thought I'd save the build cycle and rebase mine (as it is the smaller change anyway) :-)

**Pull request merged**